evcxr/src/command_context.rs

```diff
@@ -159,6 +166,55 @@ Panic detected. Here's some useful information if you're filing a bug report.
         }
     }
 
+    fn execute_shell_command(
+        &mut self,
+        shell_command: &ShellCommand,
+    ) -> Result<EvalOutputs, Error> {
+        use std::process::Command;
+
+        let command_output = Command::new("sh")
+            .arg("-c")
+            .arg(&shell_command.command)
+            .output();
+
+        match command_output {
+            Ok(output) => {
+                let stdout_str = String::from_utf8_lossy(&output.stdout);
+                let stderr_str = String::from_utf8_lossy(&output.stderr);
+
+                let mut eval_outputs = EvalOutputs::default();
+
+                // Add the stdout and stderr to eval_outputs
+                eval_outputs
+                    .content_by_mime_type
+                    .insert("stdout".to_string(), stdout_str.into());
+                eval_outputs
+                    .content_by_mime_type
+                    .insert("stderr".to_string(), stderr_str.clone().into());
+
+                if !output.status.success() {
+                    // Handle non-zero exit status
+                    let error_message = format!(
+                        "Shell command failed with exit code {}: {}",
+                        output.status.code().unwrap_or_default(),
+                        stderr_str
+                    );
+                    // Handle this error
+                    return Err(Error::from(error_message));
+                }
+
+                Ok(eval_outputs)
+            }
+            Err(error) => {
+                // Handle the case when executing the command fails
+                Err(Error::from(format!(
+                    "Failed to execute shell command: {}",
+                    error
+                )))
+            }
+        }
+    }
+
     pub fn set_opt_level(&mut self, level: &str) -> Result<(), Error> {
         self.eval_context.set_opt_level(level)
     }
```

Comment:
It can't see any way to add a comment on the appropriate line in Github, but could you also filter out shell commands in prepare_for_analysis below? At the moment, if I have a command followed by some Rust code, then I get errors reported (red squiggles) in Jupyter Notebook. This appears to be because the shell command is ending up in the code being passed to cargo check.

![image](https://github.com/evcxr/evcxr/assets/8983542/1f45ee69-090e-4eb3-99f2-7513587e3e7d)